src/posix/platform/trel_udp6.cpp

```diff
@@ -405,6 +413,102 @@ static void ReceivePacket(int aSocket, otInstance *aInstance)
     }
 }
 
+static void processNetifAddrEvent(struct nlmsghdr *aNetlinkMessage)
+{
+    struct ifaddrmsg *ifaddr = reinterpret_cast<struct ifaddrmsg *>(NLMSG_DATA(aNetlinkMessage));
+    size_t            rtaLength;
+    otError           error = OT_ERROR_NONE;
+
+    VerifyOrExit(ifaddr->ifa_index == static_cast<unsigned int>(sInterfaceIndex) && ifaddr->ifa_family == AF_INET6);
+
+    rtaLength = IFA_PAYLOAD(aNetlinkMessage);
+
+    for (struct rtattr *rta = reinterpret_cast<struct rtattr *>(IFA_RTA(ifaddr)); RTA_OK(rta, rtaLength);
+         rta                = RTA_NEXT(rta, rtaLength))
+    {
+        if (rta->rta_type != IFA_ADDRESS)
+        {
+            continue;
+        }
+
+        {
+            otIp6Address addr;
+            memcpy(&addr, RTA_DATA(rta), sizeof(addr));
+            bool isInterfaceAddress = memcmp(&sInterfaceAddress, RTA_DATA(rta), sizeof(sInterfaceAddress)) == 0;
+
+            if (!isInterfaceAddress)
+            {
+                continue;
+            }
+
+            if (aNetlinkMessage->nlmsg_type == RTM_NEWADDR)
+            {
+                otLogInfoPlat("[trel] Interface address added successfully.");
+                sAddInterfaceAddressDeadline = 0;
+            }
+            else if (aNetlinkMessage->nlmsg_type == RTM_DELADDR)
+            {
+                otLogWarnPlat("[trel] Interface address removed unexpectedly.");
```

Comment:
How about moving this log line under `if (sAddInterfaceAddressDeadline == 0)` block?

If I recall correctly, on some/most linux platforms, whenever we request an address add, we can see a `RTM_DELADDR` first then followed by a `RTM_NEWADDR` (independent of whether the address was present on netif before). In such a case, we will see this warning log all the time (even when the address is being added ok).

Reply:
Make sense to me.